pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/CountAggregationFunction.java

```diff
@@ -50,7 +50,11 @@ public CountAggregationFunction(ExpressionContext expression) {
   public CountAggregationFunction(ExpressionContext expression, boolean nullHandlingEnabled) {
     super(expression);
     // Consider null values only when null handling is enabled and function is not COUNT(*)
-    _nullHandlingEnabled = nullHandlingEnabled && !expression.getIdentifier().equals("*");
+    // Note COUNT on any literal gives same result as COUNT(*)
+    // So allow for identifiers that are not * and functions, disable for literals and *
+    _nullHandlingEnabled = nullHandlingEnabled
+            && ((expression.getType() == ExpressionContext.Type.IDENTIFIER && !expression.getIdentifier().equals("*"))
+            || (expression.getType() == ExpressionContext.Type.FUNCTION));
```

Comment:
shouldn't we also allow literal type. it looks like the test is done on COUNT(1) right?

Reply:
My thought was to give COUNT(literal) same treatment as COUNT(*) as these give same result of returning total # of rows in a collection regardless of nulls. 
But main goal of fix was to avoid the getIdentifier.equals() NPE. Should I revise the statement to allow null handling for identifiers that are not * and all non-identifiers?

Reply:
technically speaking in standardized SQL count(col) returns a number of non-null value count for that column in a table. 
- `count(*)` and `count(1)` (or count any non-null literal) are equivalent
- `count(col)` is technically not equivalent to `count(*)`
    - when nullHandlingEnabled = false. it is equivalent to `count(*)` 
    - when nullHandlingEnabled = true. i am not sure if the current result is correct (but this is out of scope of this PR)

so my suggestion is 
```suggestion
            || (expression.getType() == ExpressionContext.Type.FUNCTION)
            || (expression.getType() == ExpressionContext.Type.LITERAL && expression.getLiteral().getValue() != null));
```

Reply:
not sure if this is necessary now since we don't have null literal support properly. CC @61yao please share your comment.

Reply:
With the change the only scenario not handled is `count(null)`, which IMO is a useless operation because it always return 0. 
The fix is basically just preventing the NPE from the '*' check

Reply:
not necssarily b/c we have compile time function invoker that can produce a NULL. but agreed we can address separately.